src/handler/guild/role.ts

```diff
@@ -138,128 +130,184 @@ export class Role {
         await interaction.deferReply();
 
         const role = interaction.options.getRole("role", true);
+        const type = interaction.options.getString("type", true);
+
+        if (type === "admin") {
+            await this.adminRoleAdd(interaction, role.id);
+        } else {
+            await this.uniqueRoleAdd(interaction, role.id, type);
+        }
+    }
+
+    async adminRoleAdd(
+        interaction: CommandInteraction,
+        roleID: string
+    ): Promise<void> {
+        await getConnection()
+            .createEntityManager()
+            .transaction(async (manager) => {
+                const guildEntity = await manager.findOneOrFail(
+                    GuildEntity,
+                    interaction.guildId
```

Comment:
why are you not use the repository pattern here?

Reply:
best as I can tell that doesn't work with database transactions

Reply:
https://github.com/typeorm/typeorm/issues/1521 seems like it maybe kind of does but not really in a very useful way so i guess maybe skip it lol